File: python/claude/p134.py
from flask import Flask
from html import escape

app = Flask(__name__)

# Decorator that maps URL to function
@app.route('/hello/<username>')
def hello(username):
    """
    Decorator creates a mapping between the URL and this function.
    The username parameter is extracted from the URL.
    The escape() function converts special characters to HTML-safe sequences.
    """
    escaped_username = escape(username)
    return f"Hello {escaped_username}"

if __name__ == '__main__':
    app.run(debug=True)

File: javascript/claude/p134.js
const express = require('express');
const app = express();

// Function to escape special characters to HTML-safe sequences
function escape(str) {
    return String(str)
        .replace(/&/g, '&amp;')
        .replace(/</g, '&lt;')
        .replace(/>/g, '&gt;')
        .replace(/"/g, '&quot;')
        .replace(/'/g, '&#039;');
}

// Decorator/Route handler that maps URL to function
app.get('/hello/:username', (req, res) => {
    // The username parameter is extracted from the URL
    const username = req.params.username;
    // escape() function converts special characters to HTML-safe sequences
    const escapedUsername = escape(username);
    res.send(`Hello ${escapedUsername}`);
});

const PORT = 3000;
app.listen(PORT, () => {
    console.log(`Server is running on http://localhost:${PORT}`);
});
